aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLibBlas.cpp

```diff
@@ -218,6 +218,19 @@ static void apply_triangular_solve_batched(const Tensor& A, const Tensor& B, boo
 }
 
 void triangular_solve_batched_cublas(const Tensor& A, const Tensor& B, bool left, bool upper, TransposeType transpose, bool unitriangular) {
+  // Workaround the following a bug on CUDA < 12.1
+  // RuntimeError: CUDA error: CUBLAS_STATUS_EXECUTION_FAILED when calling `cublasStrsmBatched
+  // See https://github.com/pytorch/pytorch/issues/79191#issuecomment-1154222580
+#if defined(CUSOLVER_VERSION) && CUSOLVER_VERSION < 12100
+  if (B.size(-1) > 524280) {
+    auto n_chunks = (B.size(-1) + 524280 - 1) / 524280; // ceildiv
```

Comment:
NIt (avoid repeating the same hardcoded constant thrice)
```suggestion
  constexpr auto max_batch_size = 524280;
  if (B.size(-1) > max_batch_size) {
    auto n_chunks = (B.size(-1) + max_batch_size - 1) / max_batch_size; // ceildiv
```